workspaces/ballerina/ballerina-side-panel/src/components/editors/MultiModeExpressionEditor/ChipExpressionEditor/ChipExpressionBaseComponent.tsx

```diff
@@ -258,7 +279,15 @@ export const ChipExpressionBaseComponent = (props: ChipExpressionBaseComponentPr
         setIsCompletionsOpen(false);
     };
 
-    const handleHelperPaneValueChange = async (value: string, closeHelperpane: boolean) => {
+    const handleHelperPaneValueChange = async (updatedValue: string, closeHelperpane: boolean) => {
+        let value = updatedValue;
+        if (updatedValue.endsWith('()')) {
+            const signature = await getFnSignature(updatedValue, updatedValue.length - 1);
+            if (signature) {
+                const argsString = signature.args.map((_, index) => `$${index + 1}`).join(',');
+                value = updatedValue.slice(0, -1) + argsString + ')';
+            }
+        }
```

Comment:
The logic for handling function signatures with parameters appears in both `handleCompletionSelect` and `handleHelperPaneValueChange`. Consider extracting to a helper function.